### PR TITLE
Add info on SimpleDiscoveryClient in the docs. Remove redundant param.

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**

### CODE REVIEW

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 1]
```diff
@@ -277,6 +277,22 @@ implementations provided by Spring Cloud, among others  `ConsulDiscoveryClient`,
...
+instance, that uses properties to get information on service and instances, will be used.
+
+The information about the available instances should be passed to via properties in the following format:
+`spring.cloud.discovery.client.simple.instances.service1[0].uri=http://s11:8080`, where
+`spring.cloud.discovery.client.simple.instances` is the common prefix, then `service1` stands
+for the ID of the service in question, while `[0]` indicates the index number of the instance
+(as visible in the example, indexes start with `0`), and then the value of `uri` is
+the actual URI under which the instance is available.
+
+WARN:: `SimpleDiscoveryClient` has been added for testing and demo purposes. We do not recommend
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc]: I don't agree with this. It's also always there in a composite with the lowest priority.

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryClientAutoConfiguration.java [hunk 2]
```diff
@@ -68,7 +67,7 @@ public SimpleDiscoveryProperties simpleDiscoveryProperties() {
 	}
 
 	@Bean
-	@Order(Ordered.LOWEST_PRECEDENCE)
+	@Order
```
Comment (spencergibb) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryClientAutoConfiguration.java]: Why this change?

Comment (OlgaMaciaszek) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryClientAutoConfiguration.java]: This is not necessary as `Ordered.LOWEST_PRECEDENCE` is the default value.

File: docs/src/main/asciidoc/spring-cloud-commons.adoc (hunk 1 above)
Comment (OlgaMaciaszek) [reply docs/src/main/asciidoc/spring-cloud-commons.adoc]: Seemed to me that using a ServiceRegistry is way more production-appropriate than static input that does not track the state of the instances. However, can remove it.

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryClientAutoConfiguration.java (hunk 2 above)
Comment (spencergibb) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/SimpleDiscoveryClientAutoConfiguration.java]: so it is!

File: docs/src/main/asciidoc/spring-cloud-commons.adoc (hunk 1 above)
Comment (spencergibb) [reply docs/src/main/asciidoc/spring-cloud-commons.adoc]: I think it has uses for external systems that can't register. Let's remove the warning. Other than that the docs are great!

Comment (OlgaMaciaszek) [reply docs/src/main/asciidoc/spring-cloud-commons.adoc]: Done.
